plugins/ufw/ufwclient.cpp

```diff
@@ -507,3 +507,70 @@ bool UfwClient::logsAutoRefresh() const
 {
     return m_logsAutoRefresh;
 }
+
+RuleWrapper* UfwClient::createRuleFromConnection(const QString &protocol, const QString &localAddress, const QString &foreignAddres, const QString &status)
+{
+    auto _localAddress = localAddress;
+    _localAddress.replace("*", "");
+    _localAddress.replace("0.0.0.0", "");
+
+    auto _foreignAddres = foreignAddres;
+    _foreignAddres.replace("*", "");
+    _foreignAddres.replace("0.0.0.0", "");
+
+    auto localAddressData = _localAddress.split(":");
+    auto foreignAddresData = _foreignAddres.split(":");
+
+    auto rule = new RuleWrapper({});
+    rule->setIncoming(status == QStringLiteral("LISTEN"));
+    rule->setPolicy("deny");
+
+    // Prepare rule draft
+    if (status == QStringLiteral("LISTEN")) {
+        rule->setSourceAddress(foreignAddresData[0]);
```

Comment:
This has the potential to be crashy, we don't check the length after a string split

Reply:
This was just a copy / paste / move to C++ from Javascript code. I'll do the proper checks.